draft-bbs-signatures.md

```diff
@@ -389,78 +387,104 @@ Procedure:
 This operation computes a deterministic signature from a secret key (SK) and optionally over a header and or a vector of messages.
 
 ```
-signature = Sign(SK, PK, header, (msg_1,..., msg_L), (H_1,..., H_L))
+signature = Sign(SK, PK, header, messages)
 
 Inputs:
 
-- SK (REQUIRED), an octet string of the form outputted by the KeyGen operation.
-- PK (REQUIRED), an octet string of the form outputted by the SkToPk operation provided the above SK as input.
-- header (OPTIONAL), an octet string containing context and application specific information. If not supplied, it defaults to an empty string.
-- msg_1,..., msg_L (OPTIONAL), a vector of octet strings.
-- H_1,..., H_L (OPTIONAL), points of G1. Generators used to commit each message.
+- SK (REQUIRED), a non negative integer mod r outputted by the KeyGen
+                 operation.
+- PK (REQUIRED), an octet string of the form outputted by the SkToPk
+                 operation provided the above SK as input.
+- header (OPTIONAL), an octet string containing context and application
+                     specific information. If not supplied, it defaults
+                     to an empty string.
+- messages (OPTIONAL), a vector of octet strings. If not supplied, it
+                       defaults to the empty array "()".
 
 Parameters:
 
-- Ciphersuite_ID (REQUIRED), ASCII string. The unique ID of the ciphersuite.
-- H_s (REQUIRED), point of G1. The generator for the blinding value of the signature.
-- H_d (REQUIRED), point of G1. The generator used to sign the signature domain.
+- Ciphersuite_ID, ASCII string. The unique ID of the ciphersuite.
+- generator_seed, ASCII string. The generators seed defined by the
+                  ciphersuite
 
 Definitions:
 
-- L, is the non-negative integer representing the number of messages to be signed e.g length(msg_1,...,msg_L).  Note if no messages are supplied as an input to this operation, the value of L MUST evaluate to zero (0).
+- L, is the non-negative integer representing the number of messages to
+     be signed e.g length(messages). If no messages are supplied as an
+     input, the value of L MUST evaluate to zero (0).
 
 Outputs:
 
 - signature, a signature encoded as an octet string.
 
+Precomputations:
```

Comment:
Just a note I think we should provide an explanation some where as to why we are adding another concept to the definition of an operation. I assume this is because the result of such could be cached or handled in an alternative manner?

Reply:
The real reason is for readability. In practice it's a "setting up" process where we can get all necessary elements for the "core procedure". So essentially yes, staff that can be cached and that "cannot" return INVALID.

Added a small paragraph in the start of the Core operations section. We can always update that as well.